CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration_UI.uc

```diff
@@ -511,5 +512,57 @@ static protected function EventListenerReturn StrategyPolicy_ShowCovertActionsOn
 	// Never show actions popup after UIStrategyPolicy
 	Tuple.Data[0].b = false;
 	
+	return ELR_NoInterrupt;
+}
+
+////////////////////
+/// Tactical HUD ///
+////////////////////
+
+static function CHEventListenerTemplate CreateTacticalHUDListeners()
+{
+	local CHEventListenerTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'CHEventListenerTemplate', Template, 'Infiltration_UI_TacticalHUD');
+	Template.AddCHEvent('OverrideReinforcementsAlert', IncomingReinforcementsDisplay, ELD_Immediate);
+	Template.RegisterInTactical = true;
+
+	return Template;
+}
+
+static function EventListenerReturn IncomingReinforcementsDisplay(Object EventData, Object EventSource, XComGameState GameState, Name EventID, Object CallbackData)
+{
+	local XComGameState_AIReinforcementSpawner ReinforcementSpawner;
+	local XComLWTuple Tuple;
+	local int NextReinforcements;
+
+	Tuple = XComLWTuple(EventData);
+
+	if (Tuple == none)
+	{
+		return ELR_NoInterrupt;
+	}
+
+	foreach `XCOMHISTORY.IterateByClassType(class'XComGameState_AIReinforcementSpawner', ReinforcementSpawner)
+	{
+		if (ReinforcementSpawner.Countdown > 0)
+		{
+			if (NextReinforcements > ReinforcementSpawner.Countdown || NextReinforcements == 0)
+			{
+				NextReinforcements = ReinforcementSpawner.Countdown;
+			}
+		}
+	}
+
+	if (NextReinforcements == 0)
+	{
+		NextReinforcements = class'XComGameState_CIReinforcementsManager'.static.GetNextReinforcements();
```

Comment:
For sake of completeness you should check both spawner and manager, and use the lowest of 2